api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -349,6 +353,21 @@ type AzureManagedControlPlaneStatus struct {
 	// next reconciliation loop.
 	// +optional
 	LongRunningOperationStates Futures `json:"longRunningOperationStates,omitempty"`
+
+	// OIDCIssuerProfile is the OIDC issuer profile of the Managed Cluster.
+	// +optional
+	OIDCIssuerProfile *OIDCIssuerProfileStatus `json:"oidcIssuerProfile,omitempty"`
+}
+
+// OIDCIssuerProfileStatus is the OIDC issuer profile of the Managed Cluster.
+type OIDCIssuerProfileStatus struct {
+	// Enabled is whether the OIDC issuer is enabled.
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
```

Comment:
this is interesting

It differs a bit from existing patterns we have for setting "read-only" fields (eg. kubelet identity). This is arguably better and more aligned with k8s pattern (spec is desired state and status is read-only actual state) but inconsistent with other fields we already have. I'm not sure about duplicating the enabled field in both spec and status, that strikes me as redundant.

What was the reasoning in including the status field?

Reply:
Overall this was done in the spirit of #3855. ASO's ManagedCluster type is also structured the same way. One concrete case that having `enabled` in the status helps with is the webhook validation to return an error when a user tries to disable the OIDC issuer, which is not allowed by AKS.

The field itself is passed through verbatim from CAPZ to the AKS API, where `nil` means "don't change" and non-nil `true` and `false` set it explicitly. That means when `enabled` is nil in the AzureManagedControlPlane spec, that by itself is not enough to know whether or not the OIDC issuer is actually enabled at that moment. Since AKS (always, AFAICT) populates `enabled` in its API when the user sends `nil`, propagating that to the status in AzureManagedControlPlane lets the webhook definitively know the actual state of `enabled` when validating an update.

Reply:
What's the advantage of letting the user set the `spec` field to "nil"? Would the same enforcement be covered by a simpler immutable check (once it's set to true, cannot be unset or set to false)?

Reply:
Allowing `nil` means users that have already set this on their clusters via other means won't have this change out from under them when CAPZ upgrades. I believe @mtougeron expressed this kind of concern a while ago around the autoscaler settings.

Reply:
agreed that the consistency would be nice. I can't remember though with the cluster-autoscaler if this was due to the way that Azure handled the nil values when passed or if it was because I didn't have a way to "reset" to the default values via an API when the user unset the field in CAPZ. Having this flag would make it more about CAPZ managing the resource/fields and that problem would pretty much go away.

Reply:
Opened #3990 to track this.

Reply:
And if `spec.oidcIssuerProfile` is null in the spec, should that always imply `spec.oidcIssuerProfile.enabled` is false?

Reply:
@nojnhuh, no the danger with using a plain non-pointer is that the default could change and then we wouldn't know if a user is intentionally disabling it or not setting the field.

To summarize there are two options/patterns possible:

A) The spec field is an optional pointer bool and defaults the value to a sane default when nil. We don't support users setting things outside of the spec, the CAPZ spec is the only source of truth for the object. This is what we do for most fields if not all currently.
B) The spec field is an optional pointer bool, we don't default the value and let "nil" be a valid value to pass to the AKS API. We use a status field to track the "actual" value in Azure. This allows users to set this value outside of the CAPZ spec and not get overwritten, as long as AKS doesn't default/require a value. This is what this PR is doing currently, and what ASO does.

IMO both are valid. I tend to prefer consistency overall because that is less likely to lead users to make mistakes. This is not a hill I'm willing to die on (and I really want to see this PR get merged so I don't want to be standing in the way of that) but my personal preference would be to keep A) for now for consistency with the agreement that B) is what we're trying to evolve towards with a v2 of the API that wraps around ASO types (https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/3629).

Reply:
> my personal preference would be to keep A) for now for consistency with the agreement that B) is what we're trying to evolve towards with a v2 of the API that wraps around ASO types

Agree this seems like the best path for now. I'll update this.

Reply:
@CecileRobertMichon This should work the way you described now. Can we still get this in for the v1.11 release?